### ISSUE TITLE
Selection checkbox, radio buttons and detail pane buttons are unlabeled or doesn't have unique label.

### ISSUE THREAD

**Issue by chlane** [vmware/clarity#4669]
**Clarity Version:** 3.1.1
**WCAG 2.1 Guideline:** 4.1.2 Name, Role, Value

**Test Environment:**
- Test URL: https://clarity.design/documentation/datagrid/
- OS: Windows 10
- User Agents/Assistive Technologies: Firefox 76.0
- Screen reader: NVDA 2020.1

**Steps to reproduce:**
1. Open the test URL.
2. Navigate through different datagrids, which has different selection options provided for the rows in the table. Like Checkbox, radio buttons or Detail pane button and observe the screen reader&#039;spoken feedback.

**Expected result:**
Appropriate unique label should be announced to the controls, so that they don&#039;t lose it&#039;s context when screen reader user access controls directly using button shortcuts. It will also help speech input users.

**Actual result:**
All the data row checkbox label is "Select". Similar issue is observed for radio buttons and detail pane button.

Reported by @ghkadK

**Comment by chlane** [comment]
The non-unique checkbox labels can be seen in other datagrid variations as well. See https://files.slack.com/files-pri/T8YCM8N9L-F016YTEHJLD/screen_shot_2020-07-13_at_9.32.26_am.png

**Comment by gkhadK** [comment]
We can either use a static label using aria-label or we can use aria-labelledby attribute to use concatenated label. i.e. use one of the columns header text and unique value from that cell corresponding to that column/row. For e.g. User ID column id=123 with value of "gkhadke" with id=1234. Using aria-labelledbv attribute for checkbox as aria-labelledby="123 1234" will announce the label as "User ID gkhadke". This solution will work, if the data in the column is unique and easily recognized by the users.

**Comment by Deepapillai88** [comment]
(P1) 20 Update/Patches TAB - View Module
in one of the bug, they have mentioned to provide the interactive text contents , for the image/icons,  or if the image is decorative then to add aria-hidden.

in the clr-tree-node the caret icon is added implicitly inside the clarity component , so not possible to add any of the attributes for the caret icon or hide it.
 need to fix this issue in the <clr-tree-node> (collapse / expand)

**Comment by Deepapillai88** [comment]
Security TAB - View Module
 In the Bug Report they have mentioned to provide unique name for interactive elements, when i check all the checkbox in the grid , it read 'select checkbox' for all checkbox which is not unique as mentioned in bug, but when i tried to provide unique name for the check box , i found those are coming in the clarity grid , so i believe check box are coming inside the clarity grid , so unable to provide unique value explicitly ..
